src/core/connection.c

```diff
@@ -3059,6 +3059,14 @@ QuicConnProcessPeerTransportParameters(
             Connection->Stats.GreaseBitNegotiated = TRUE;
         }
 
+        if (Connection->Settings.ReliableResetEnabled &&
+            (Connection->PeerTransportParams.Flags & QUIC_TP_FLAG_RELIABLE_RESET_ENABLED) > 0) {
+            //
+            // Reliable Reset is enabled on both sides.
+            //
+            Connection->State.ReliableResetStreamNegotiated = TRUE;
+        }
+
```

Comment:
Right here, if `Connection->Settings.ReliableResetEnabled` is set, you should then deliver the new event to the app passing `Connection->State.ReliableResetStreamNegotiated` to them.

Reply:
Added call to callback.

Reply:
You put the callback above, not on this line. We need to give this callback regardless of if the feature was negotiated. It needs to be delivered if the app set `Connection->Settings.ReliableResetEnabled` to `TRUE`.

Reply:
Thanks for the clarification!
We always deliver callback if the app set local settings to TRUE. 
I can move callback out of the negotiation branch.